pkg/kubefed2/federate/federate.go

```diff
@@ -308,12 +329,30 @@ func CreateFedResource(hostConfig *rest.Config, resources *fedResources, dryrun
 
 	qualifiedFedName := ctlutil.NewQualifiedName(resources.FedResource)
 	if !dryrun {
-		_, err = fedClient.Resources(resources.FedResource.GetNamespace()).Create(resources.FedResource, metav1.CreateOptions{})
-		if err != nil {
-			return errors.Wrapf(err, "Error creating %s %q", fedKind, qualifiedFedName)
-		}
+		// It might take a little while for the federated type to appear if the
+		// same is being enabled while or immediately before federating the resource.
+		err = wait.PollImmediate(createResourceRetryInterval, createResourceRetryTimeout, func() (bool, error) {
+			_, err := fedClient.Resources(resources.FedResource.GetNamespace()).Create(resources.FedResource, metav1.CreateOptions{})
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			if err != nil {
+				return false, err
+			}
+			return true, nil
+		})
```

Comment:
Where is the check for an error condition?

Reply:
Added!